plugin/plugin.go

```diff
@@ -67,51 +71,66 @@ type Args struct {
 
 	// connect hostname (required)
 	ConnectHostname string `envconfig:"PLUGIN_CONNECT_HOSTNAME"`
+	// Issue Keys(optional)
+	IssueKeys []string `envconfig:"PLUGIN_ISSUEKEYS"`
 }
 
 // Exec executes the plugin.
 func Exec(ctx context.Context, args Args) error {
 	var (
-		environ  = toEnvironment(args)
-		issue    = extractIssue(args)
-		state    = toState(args)
-		version  = toVersion(args)
-		deeplink = toLink(args)
+		environ         = toEnvironment(args)
+		environmentID   = toEnvironmentId(args)
+		environmentType = toEnvironmentType(args)
+		issues          []string
+		state           = toState(args)
+		version         = toVersion(args)
+		deeplink        = toLink(args)
 	)
 
+	// ExtractInstanceName extracts the instance name from the provided URL if any
+	instanceName := ExtractInstanceName(args.Instance)
+
 	logger := logrus.
 		WithField("client_id", args.ClientID).
 		WithField("cloud_id", args.CloudID).
 		WithField("project_id", args.Project).
-		WithField("instance", args.Instance).
+		WithField("instance", instanceName).
 		WithField("pipeline", args.Name).
 		WithField("environment", environ).
 		WithField("state", state).
-		WithField("version", version)
+		WithField("environment Type", environmentType).
+		WithField("environment ID", environmentID)
 
-	if issue == "" {
-		logger.Debugln("cannot find issue number")
-		return errors.New("failed to extract issue number")
+	//check if PLUGIN_ISSUEKEYS is provided
+	if len(args.IssueKeys) > 0 {
+		issues = args.IssueKeys
+	} else {
+		// fallback to extracting from commit if no issue keys are passed
+		var issue string = extractIssue(args)
+		if issue == "" {
+			logger.Debugln("cannot find issue number")
+			return errors.New("failed to extract issue number")
+		}
+		issues = []string{issue} // add the single issue here for consistency
 	}
 
 	commitMessage := args.Commit.Message
-    	if len(commitMessage) > 255 {
-        	logger.Warnln("Commit message exceeds 255 characters; truncating to fit.")
-        	commitMessage = commitMessage[:252] + "..."
-    	}
+	if len(commitMessage) > 255 {
+		logger.Warnln("Commit message exceeds 255 characters; truncating to fit.")
+		commitMessage = commitMessage[:252] + "..."
+	}
 
-	logger = logger.WithField("issue", issue)
 	logger.Debugln("successfully extraced issue number")
-
 	deploymentPayload := DeploymentPayload{
 		Deployments: []*Deployment{
 			{
 				Deploymentsequencenumber: args.Build.Number,
 				Updatesequencenumber:     args.Build.Number,
+				IssueKeys:                issues,
 				Associations: []Association{
 					{
 						Associationtype: "issueIdOrKeys",
-						Values:          []string{issue},
+						Values:          issues,
```

Comment:
let's just pass it under associations as suggested in the Jira API doc:
- When PLUGIN_ISSUE_KEYS is provided:
```
"associations": [
    {
        "associationType": "issueIdOrKeys",
        "values": [
            "ABC-123",
            "ABC-456"
        ]
    }
]
```
- When PLUGIN_ISSUE_KEYS is not provided (and an issue key is extracted):
```
"associations": [
    {
        "associationType": "issueIdOrKeys",
        "values": [
            "ABC-123"
        ]
    }
]
```